### PR TITLE
Working register and corrected routers

### PR DESCRIPTION
**Pull request by VadymPolianskyi**

### CODE REVIEW

File: softgroup/basis/common-datamapper/src/main/java/com/softgroup/common/datamapper/JacksonDataMapper.java [hunk 1]
```diff
@@ -9,14 +9,17 @@
...
 
 import java.io.InputStream;
 import java.util.Map;
 
 /**
  * @author Arthas
  */
+@Component
 public class JacksonDataMapper implements DataMapper {
+	//@Import({value=DataMapperAppCfg.class})
```
Comment (Dominys) [inline softgroup/basis/common-datamapper/src/main/java/com/softgroup/common/datamapper/JacksonDataMapper.java]: @VadymPolyanski remove commented code.

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 2]
```diff
@@ -7,24 +7,31 @@
 import com.softgroup.common.protocol.Response;
 import com.softgroup.common.protocol.ResponseData;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
+import java.lang.reflect.ParameterizedType;
 import java.util.Map;
-
+@Component
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @VadymPolyanski How you suppose to create component from abstract class ?

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 3]
```diff
@@ -7,24 +7,31 @@
...
 public abstract class AbstractRequestHandler<T extends RequestData, R extends ResponseData> implements RequestHandler {
 
     @Autowired
     JacksonDataMapper mapper;
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java]: @VadymPolyanski Interface!

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java [hunk 4]
```diff
@@ -7,24 +7,31 @@
...
     public abstract Response<R> doHandle(Request<T> request);
 
     @Override
     public Response<R> handle(Request<?> msg) {
 
-        Map<String, Object> map = mapper.convertToMap(msg);
-        T data = mapper.convert(map, (Class<T>) RequestData.class);
+        Map<String, Object> map = (Map<String, Object>) msg.getData();
+
+        Class<T> persistentClass = (Class<T>)
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java:26]: @VadymPolyanski Move this to separate field

File: softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java [hunk 5]
```diff
@@ -4,15 +4,16 @@
 import com.softgroup.common.protocol.Response;
 import com.softgroup.common.router.api.factory.HandlerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 
-public abstract class AbstractRouterHandler<T extends Handler> implements RouterHandler, CommonRouterHandler {
+@Component
+public abstract class AbstractRouterHandler<T extends AbstractRequestHandler> implements RouterHandler, CommonRouterHandler {
```
Comment (Dominys) [inline softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java]: @VadymPolyanski Abstract component ??

File: softgroup/frontend/testing/pom.xml [hunk 6]
```diff
@@ -7,9 +7,69 @@
         <groupId>com.softgroup</groupId>
         <version>0.0.1</version>
     </parent>
+
```
Comment (Dominys) [inline softgroup/frontend/testing/pom.xml]: @VadymPolyanski What is package testing? And why you need separate module ??

File: softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/configuration/RestApplicationConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * Author: vadym
+ * Date: 01.03.17
+ * Time: 17:47
+ */
+
+@Configuration
+@ComponentScan(basePackages = {
+        "com.softgroup"})
```
Comment (Dominys) [inline softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/configuration/RestApplicationConfiguration.java]: @VadymPolyanski Scan only this module.

File: softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/configuration/WebConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Author: vadym
+ * Date: 01.03.17
+ * Time: 17:54
+ */
+@Configuration
+@EnableWebMvc
+@ComponentScan(basePackages = {
+        "com.softgroup"})
```
Comment (Dominys) [inline softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/configuration/WebConfiguration.java]: @VadymPolyanski Scan only this module

File: softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/controller/MessengerController.java [hunk 9]
```diff
@@ -1,9 +1,41 @@
...
+    DataMapper jacksonDataMapper;
+
+    @RequestMapping(path = "/registration",
+            method = RequestMethod.POST)
+    public Response publicMessage(@RequestBody final String requestStr) {
+        CommonRequest request = jacksonDataMapper.mapData(requestStr.getBytes(StandardCharsets.UTF_8),CommonRequest.class);
+        return firstRouterHandler.handle(request);
+    }
+
+    @RequestMapping(path = "/test",
```
Comment (Dominys) [inline softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/controller/MessengerController.java]: @VadymPolyanski ?

File: softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/controller/MessengerController.java [hunk 10]
```diff
@@ -1,9 +1,41 @@
...
+@RestController
+@RequestMapping(path = "/messenger")
 public class MessengerController {
+    @Autowired
+    Handler firstRouterHandler;
+
+    @Autowired
+    DataMapper jacksonDataMapper;
+
+    @RequestMapping(path = "/registration",
```
Comment (Dominys) [inline softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/controller/MessengerController.java]: @VadymPolyanski Why registration ?

Comment (VadymPolianskyi) [reply softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/controller/MessengerController.java]: @Dominys It was just test)

File: softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/configuration/RestApplicationConfiguration.java (hunk 7 above)
Comment (VadymPolianskyi) [reply softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/configuration/RestApplicationConfiguration.java]: @Dominys have done

File: softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/configuration/WebConfiguration.java (hunk 8 above)
Comment (VadymPolianskyi) [reply softgroup/frontend/testing/src/main/java/com/softgroup/frontend/test/configuration/WebConfiguration.java]: @Dominys have done

**Pull request merged**